content/en/blog/_posts/2025-XX-XX-nftables-kube-proxy/index.md

```diff
@@ -0,0 +1,238 @@
+results!:
+
+![kube-proxy iptables-vs-nftables first packet latency, at various percentiles, in clusters of various sizes](iptables-vs-nftables.png)
+
+## Why NFTables? Part 2: Control Plane Latency
+
+While the improvements to data plane latency in large clusters are
+great, there's another problem with iptables kube-proxy that often
+keeps users from even being able to grow their clusters to that size:
+the time it takes kube-proxy to program new iptables rules when
+services and endpoints change.
+
+With both iptables and nftables, the total size of the "ruleset" as a
+whole (actual "rules", plus associated data) is **O(n)** in the combined
+number of services and endpoints. Originally, the iptables backend would
+rewrite every rule on every update, and with tens of thousands of
+services, this could grow to be hundreds of thousands of iptables rules.
+Starting in Kubernetes 1.26, we began improving kube-proxy so that it
+could skip updating _most_ of the unchanged rules in each update, but
+the limitations of `iptables-restore` as an API meant that it was still
+always necessary to send an update that's **O(n)** in the number of
+services (though with a noticeably smaller constant than it used to be).
+Even with those optimizations, it can still be necessary to make use of
+kube-proxy's `minSyncPeriod` config option to ensure that it doesn't
+spend every waking second trying to push iptables updates.
+
+The nftables APIs allow for doing much more incremental updates, and
+when kube-proxy in nftables mode does an update, the size of the
+update is only **O(n)** in the number of _changed_ services and
+endpoints. The fact that the nftables API allows each nftables-using
+component to have its own private table also means that there is no
+global lock contention between components like with iptables. As a
+result, kube-proxy's nftables updates can be done much more
+efficiently than with iptables.
+
+(Unfortunately I don't have cool graphs for this part.)
+
+## Why _not_ NFTables?
+
+All that said, there are a few reasons why you might not want to jump
+right into using the nftables backend for now.
+
+First, while kube-proxy's nftables mode behaves very similarly to the
+iptables mode at a high level (since both APIs speak to the same
+underlying kernel subsystem), it's still possible that some existing
+network plugins that work with kube-proxy in iptables mode might not
+work right with it in nftables mode; you should consult the
+documentation (or forums, bug tracker, etc.) for your network plugin to
+see if it supports nftables mode. (Note that some plugins may not need
+any changes; as long as they don't try to directly interact with or
+override kube-proxy's iptables rules, they shouldn't care whether
+kube-proxy is using iptables or nftables.)
+
+Second, the nftables mode will not work on older Linux distributions;
+currently it requires a 5.13 or newer kernel. Additionally, because of
```

Comment:
I see , you have it here https://github.com/kubernetes/website/pull/49393/files#r1913406303